dawn/src/dawn/AST/Offsets.h

```diff
@@ -20,11 +20,19 @@
 #include <memory>
 #include <vector>
 
+namespace dawn::iir {
+class HorizontalExtent;
```

Comment:
forward declaration not needed, see https://en.cppreference.com/w/cpp/language/friend example after 3)

Reply:
with `iir::HorizontalExtent`, this does not work, because it is in a different namespace. First, it will complain that the namespace does not exist; if you introduce this namespace, it will still complain that the class does not exist.